datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -767,6 +769,71 @@ private Future<Boolean> dispatchAssignmentChangeIfNeeded(String connectorType, L
     return null;
   }
 
+  private List<DatastreamTask> getRemovedTasks(List<DatastreamTask> newAssignment, List<DatastreamTask> oldAssignment) {
+    List<DatastreamTask> removedTasks = new ArrayList<>(oldAssignment);
+    removedTasks.removeAll(newAssignment);
```

Comment:
Isn't this an expensive operation? Since the code does similar searches multiple times, should you convert it to set?

Reply:
This was copied over from the original code (i.e. code that calculates removed/added tasks). I did a conversion to HashSet to address this, but I think a better solution is to refactor this code. Most of these types can be converted to `Collection<DatastreamTask>` unless there's a specific need to use Set/List APIs. That will reduce the overall number of conversions in the code. It's out of scope from this PR, hence not touching it here.